src/main/java/htsjdk/samtools/SAMFileWriterFactory.java

```diff
@@ -372,32 +374,35 @@ private SAMFileWriter initWriter(final SAMFileHeader header, final boolean preso
      */
     public SAMFileWriter makeSAMOrBAMWriter(final SAMFileHeader header, final boolean presorted, final File outputFile) {
         final String filename = outputFile.getName();
-        if (filename.endsWith(BamFileIoUtils.BAM_FILE_EXTENSION)) {
-            return makeBAMWriter(header, presorted, outputFile);
-        }
-        if (filename.endsWith(".sam")) {
+
+        if (SAM_TYPE.hasValidFileExtension(filename)) {
             return makeSAMWriter(header, presorted, outputFile);
+        } else {
+            if (!BAM_TYPE.hasValidFileExtension(filename)) {
+                log.warn("Unknown file extension, assuming BAM format when writing file: " + outputFile.getAbsolutePath());
+            }
```

Comment:
I think that users will be need to get used to this new warning (why am I seeing this?), and they will see it every time they pipe to the output, which is used a lot.  I don't see other implementations having users asking for a warning when piping to stdout.  If you feel very strongly about this, then OK, but ultimately it's up to the new maintainers of htsjdk to decide.